transaction-impl/src/main/java/com/example/auction/transaction/impl/TransactionEntity.java

```diff
@@ -28,25 +33,66 @@ public Behavior initialBehavior(Optional<TransactionState> snapshotState) {
     private Behavior notStarted(TransactionState state) {
         BehaviorBuilder builder = newBehaviorBuilder(state);
 
-        builder.setCommandHandler(StartTransaction.class, (start, ctx) ->
-                ctx.thenPersist(new TransactionStarted(entityUUID(), start.getTransaction()), (e) ->
+        builder.setCommandHandler(StartTransaction.class, (cmd, ctx) ->
+                ctx.thenPersist(new TransactionStarted(entityUUID(), cmd.getTransaction()), (e) ->
                         ctx.reply(Done.getInstance())
                 )
         );
 
-        builder.setEventHandlerChangingBehavior(TransactionStarted.class, started ->
-                negotiatingDelivery(TransactionState.start(started.getTransaction()))
+        builder.setEventHandlerChangingBehavior(TransactionStarted.class, event ->
+                negotiatingDelivery(TransactionState.start(event.getTransaction()))
         );
 
+        addGetTransactionHandler(builder);
+
         return builder.build();
     }
 
     private Behavior negotiatingDelivery(TransactionState state) {
         BehaviorBuilder builder = newBehaviorBuilder(state);
+
+        builder.setReadOnlyCommandHandler(StartTransaction.class, (cmd, ctx) ->
+                ctx.reply(Done.getInstance())
+        );
+
+        builder.setCommandHandler(SubmitDeliveryDetails.class, (cmd, ctx) -> {
+            if(cmd.getUserId().equals(state().getTransaction().get().getWinner())) {
+                return ctx.thenPersist(new DeliveryDetailsSubmitted(entityUUID(), cmd.getDeliveryData()), (e) ->
+                        ctx.reply(Done.getInstance())
+                );
+            }
+            else
+                throw new Forbidden("Only the auction winner can submit delivery details");
```

Comment:
Changing this will break your tests.
Also, I've just noticed there's a similar case few lines below this.